File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 4/29/2026, 10:29:55 AM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max = 0
        count = 0
        for i in range(len(nums)):
            if nums[i] == 1:
                if i == 0:
                    count = 1
                else:
                    if nums[i-1] == 1:
                        count += 1
                    else:
                        count = 1
                if count > max:
                    max = count
        return max

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.js
// Last updated: 4/29/2026, 10:29:59 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
// 2022-12-06
// 2:54s
// O(n)
// O(1)
var findMaxConsecutiveOnes = function(nums) {
    let max = 0;
    let count = 0;
    for(const num of nums) {
        if(num === 1) {
            count++;
        } else {
            count = 0;
        }
        if(count > max) {
            max = count;
        }
    }
    return max;
};